andela/travel_tool_front
# #162122460 E2E: User can only edit an open request

#### What does this PR do?
- Verify that only open requests can be edited and that the edit shows up on the requests table 

#### Description of Task to be completed?
- Add end to end tests that verify:
  - the edit request modal is displayed, pre-filled with the relevant request's information
  - only open requests can be edited and that the edit shows up on the requests table
  - errors are displayed and the submit button is disabled when the required fields are empty

#### How should this be manually tested?
- Clone the repo
    `git clone https://github.com/andela/travel_tool_front.git`
 - Switch to the project directory
   `cd travel_tool_front`
 - Checkout the **ch-e2e-edit-open-request-162122460** branch
   `git checkout ch-e2e-edit-open-request-162122460`
 - Install dependencies and start the server
   - `make start` *if you have docker*
   - `yarn install` then `yarn start` *if you do not have docker*
 - Start the [backend](https://github.com/andela/travel_tool_back) of this project.
 - Create a user with the role of a manager in the database
 - Start cypress with the browser UI:
    `yarn end2end`
 - Click on **edit-request.spec.js**
 - The tests should run without failing

#### Any background context you want to provide?
 - You will need the backend on to run the end to end tests.
 - You will need a manager in the database for the tests to run without failing

#### What are the relevant pivotal tracker stories?
* [Starts #162122460](https://www.pivotaltracker.com/story/show/162122460)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
